src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -1189,16 +1279,20 @@ private IndexWriter CreateIndexWriter()
             // ensure that it's unlocked here in that case.
             try
             {
-                if (IndexWriter.IsLocked(dir))
+                if (IndexWriter.IsLocked(dir) && (!(dir is ExamineDirectory) ||
+                                                  dir is ExamineDirectory examineDirectory &&
+                                                  !examineDirectory.IsReadOnly))
                 {
+                    //TODO: if readonly index, unlock the local index?
```

Comment:
I would assume so yes but the IndexWriter in that case should already just be the local index?